### PR TITLE
[4.5] preparation

### PR DESCRIPTION
**Pull request by sbaier1**
review notes:
- the unsafe sysctl deployment template is just an alternative version of the default template with more sysctls in the security context.
- the admission hook files are adapted from [prometheus operator](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/templates/prometheus-operator/admission-webhooks/job-patch)

### CODE REVIEW

File: charts/hivemq-operator/values.yaml [hunk 1]
```diff
@@ -3,18 +3,48 @@ global:
...
+    patch:
+      enabled: true
+      image:
+        repository: jettech/kube-webhook-certgen
+        tag: v1.5.1
+        sha: ""
+        pullPolicy: IfNotPresent
+      resources: {}
+      ## Provide a priority class name to the webhook patching job
+      ##
```
Comment (schaebo) [inline charts/hivemq-operator/values.yaml:42]: empty comment line

File: charts/hivemq-operator/values.yaml [hunk 2]
```diff
@@ -199,12 +240,11 @@ hivemq:
         <cluster>
             <enabled>true</enabled>
             <transport>
-                <udp>
+                <tcp>
```
Comment (schaebo) [inline charts/hivemq-operator/values.yaml:243]: this should also be mentioned in the upgrade guide

**Review by schaebo — APPROVED** [review]

File: charts/hivemq-operator/values.yaml [hunk 3]
```diff
@@ -3,18 +3,48 @@ global:
...
     pspAnnotations: {}
     allowedCapabilities: []
+    # Image pull secrets for operator, hivemq or other images.
+    imagePullSecrets: []
+    # - name: hivemq-pull-secret
 operator:
   # Deploy a custom resource based on the hivemq section below. Set to false if you want to create a HiveMQCluster object yourself.
   # By setting this to false, the operator will not start a HiveMQ cluster when deployed.
   deployCr: true
   image: hivemq/hivemq-operator:4.4.3
```
Comment (Florian-Limpoeck) [inline charts/hivemq-operator/values.yaml:19]: is this still true?

File: charts/hivemq-operator/values.yaml [hunk 4]
```diff
@@ -199,12 +240,11 @@ hivemq:
         <cluster>
             <enabled>true</enabled>
             <transport>
-                <udp>
+                <tcp>
                     <bind-address>${HIVEMQ_BIND_ADDRESS}</bind-address>
                     <bind-port>${HIVEMQ_CLUSTER_PORT}</bind-port>
-                    <multicast-enabled>false</multicast-enabled>
                     <enabled>true</enabled>
```
Comment (Florian-Limpoeck) [inline charts/hivemq-operator/values.yaml:246]: there is no enabled property

**Review by Florian-Limpoeck — APPROVED** [review]

File: charts/hivemq-operator/values.yaml (hunk 2 above)
Comment (sbaier1) [reply charts/hivemq-operator/values.yaml:243]: i added an upgrade guide note in the ticket, is that sufficient?

File: charts/hivemq-operator/values.yaml (hunk 4 above)
Comment (sbaier1) [reply charts/hivemq-operator/values.yaml:246]: interesting, didn't know that :D

**Pull request merged**